src/lib/nvme_pt_ioctl.c

```diff
@@ -2497,3 +2500,29 @@ int opal_write_shadow_mbr_pt(struct sed_device *dev, const struct sed_key *key,
 	return ret;
 }
 
+int opal_block_sid_pt(struct sed_device *dev, bool hw_reset)
+{
+	struct opal_device *device = dev->priv;
+	int ret, req_size = BLOCKSID_PAYLOAD_SZ;
+
+	/* Send Block SID authentication command, no IF_RECV response is expected */
+
+	if (device->req_buf_size < BLOCKSID_PAYLOAD_SZ) {
```

Comment:
req_buf_size -- is guaranteed to be at min 2048B in size. I don't think that this condition will ever be true

Reply:
I prefer to leave it there in case the underlying assumption changes.

Reply:
I have just checked "Architecture Core spec" and it states in 5.2.2.4 that communication minimum is 1024B, instead of 2048B like I orginally mentioned. because of that this assumption will not change.

Reply:
Your reading of the core spec is incorrect, the minimum requirement under 5.2.2.4 suggests the minimum set of property name/value TPer and Host shall support, which defines MaxComPacketSize as 1024 for example. I double checked with the content expert on this. For BlockSID, you can in fact just have one byte, 512 byte, or 10224 bytes. They should be all fine.

Reply:
I didn't interpret spec incorrectly. I'm saying that req_buf_size is guaranteed in sedcli to be 1024 which corresponds to value from "communication mimimums" table.
Your if statement will always be false -- thus I want you to remove it.